### ISSUE TITLE
[Accessibility] Source Tree should have meaningful semantics.

### ISSUE THREAD

**Issue by yzen** [devtools-html/debugger.html#4067]
Source list tree should have tree-like semantics in order to be meaningful for screen reader users. Each row/node in the tree should be semantically a tree item with a correct depth level set.

- [ ] Expanded/collapsed state of the tree item should also be implemented using [ARIA][ar]
- [ ] Icons that represent file/folder/domain should have a text alternative. They seem to carry some semantically relevant information (whether the item is a folder or a file) so just hiding the icon from screen reader is not desirable.

---

Details about how to use ARIA to set tree semantics can be found [here]. Some general information on how to make tree-like widgets accessible can be found [here][here2].

IMPORTANT: it looks like a tree component is used to render the tree so then the component (in case it's shared) should be made accessible. This will help in places where it is re-used elsewhere. A shared tree component was already made accessible in DevTools module in mozilla-central so in case it is the same thing, we should migrate accessibility features from m-c here.

[AR]: https://www.w3.org/TR/wai-aria/states_and_properties#aria-expanded
[here]: https://www.w3.org/TR/wai-aria/roles#tree
[here2]:https://www.levelaccess.com/creating-an-accessible-aria-tree-control/

### Screenshot

![Screen Shot 2017-09-18 at 8.09.39 PM.png](https://shipusercontent.com/b2110bd75f68b14403d5358d0ec63220/Screen%20Shot%202017-09-18%20at%208.09.39%20PM.png)

**Comment by jasonLaster** [comment]
The tree comes from [here]

[here]: https://github.com/devtools-html/devtools-core/blob/master/packages/devtools-components/src/tree.js#L1

You can see a bunch of examples by running `npm storybook` or `npm test`.

**Comment by yzen** [comment]
Looks like the tree component mentioned above is up to date enough and implements many accessibility improvements from mozilla-central's devtools. 
If local tree component could be updated to the one in devtools-core it would pretty much resolve this bug. 
One serious blocked for just doing that, however, is an issue I just filed https://github.com/devtools-html/devtools-core/issues/695. Ideally it should be handled first before syncing the component.

**Comment by jasonLaster** [comment]
Great, I or @nchevobbe will do the update soon.

**Comment by HeidiEllis** [comment]
Doing Bug Triage. Can this issue be closed?

**Comment by nchevobbe** [comment]
Yes, I think we have the correct attributes now
